### PR TITLE
Throw `ArgumentException` not `ArgumentNullException` for empty string

### PR DESCRIPTION
**Pull request by xtqqczze**
cc: @iSazonov

### CODE REVIEW

**Comment by xtqqczze** [comment]
`runtime/dotnet` follows this pattern of throwing separate `ArgumentException` and `ArgumentNullException` exceptions (most of the time): https://github.com/dotnet/runtime/blob/efd818b90d61a81b73b20f3d1720279f993faa2c/src/libraries/System.Private.CoreLib/src/System/IO/File.cs#L69

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/EnableDisableRunspaceDebugCommand.cs [hunk 1]
```diff
@@ -59,7 +59,15 @@ public sealed class PSRunspaceDebug
...
         {
-            if (string.IsNullOrEmpty(runspaceName)) { throw new PSArgumentNullException(nameof(runspaceName)); }
+            if (runspaceName is null)
+            {
+                throw new PSArgumentNullException(nameof(runspaceName));
+            }
+
+            if (runspaceName.Length == 0)
+            {
+                throw new PSArgumentException(nameof(runspaceName));
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/EnableDisableRunspaceDebugCommand.cs:69]: PSArgumentException doesn’t have a constructor that only takes a parameter name.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]
Reviewed on my phone, so didn’t get to all changes. When using a ‘*ArgumentException’, please make sure it has the overload constructor you meant to use.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/EnableDisableRunspaceDebugCommand.cs (hunk 1 above)
Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/EnableDisableRunspaceDebugCommand.cs]: Fixed in 42fb50d

File: src/Microsoft.PowerShell.Commands.Management/commands/management/SetContentCommand.cs [hunk 2]
```diff
@@ -24,11 +24,16 @@ public class SetContentCommand : WriteContentCommandBase
...
         {
-            if (paths == null || paths.Length == 0)
+            if (paths is null)
             {
                 throw PSTraceSource.NewArgumentNullException(nameof(paths));
             }
 
+            if (paths.Length == 0)
+            {
+                throw PSTraceSource.NewArgumentException(null, nameof(paths));
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/SetContentCommand.cs:34]: `PSTraceSource.NewArgumentException` has an overload that takes a parameter name only, and we should use that one.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Management/commands/management/SetContentCommand.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/SetContentCommand.cs:34]: We seems need AI 😸

**Comment by iSazonov** [comment]
Perhaps we should have used named parameters to avoid confusion with these many API varieties.

**Comment by daxian-dbw** [comment]
@xtqqczze Sorry for the delay. I reviewed the changes again today, and think the pattern change may not be really necessary. Event .NET code base is not consistent on this pattern, see https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Microsoft.Extensions.DependencyModel/src/Library.cs#L43-L54.

But other maintainers may have different opinions, so I marked this PR to be reviewed by maintainers and will see what other folks think.

**Comment by iSazonov** [comment]
> @xtqqczze Sorry for the delay. I reviewed the changes again today, and think the pattern change may not be really necessary. Event .NET code base is not consistent on this pattern, see https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Microsoft.Extensions.DependencyModel/src/Library.cs#L43-L54.
> 
> But other maintainers may have different opinions, so I marked this PR to be reviewed by maintainers and will see what other folks think.

I guess new .Net pattern is a result their nullable annotation process. The Microsoft.Extensions.DependencyModel is not still annotated.

If we use the new pattern I'd suggest to use named arguments to make the code more readable.

**Comment by adityapatwardhan** [comment]
Maintainers believe that there is not enough improvement in the user experience due to this change. We recommend not taking this. Maintainers are fine with using this pattern for new code, but don't see a substantial benefit updating current code as compared to the risk of breaking change.

**Pull request closed without merging**